### PR TITLE
offsetguess: accept connections

### PR DESCRIPTION
**Pull request by iaguis**
The server was only listening and not accepting, so we were relying on
the incoming buffer to fit all the connections.

This might cause connection timeouts when the buffer fills, so we now
accept the connections and close them as soon as possible.

To avoid possible races where the connecting side is much faster than
the accepting side and can potentially fill the buffer too, we signal
from the main thread when we are about to make a connection to make sure
the server is ready to accept.

### CODE REVIEW

**Review by alban — APPROVED** [review]

**Comment by alban** [comment]
The symptoms were:

failed to init perf map for IPv4 events: error guessing offsets: error dialing "127.0.0.2:46865": dial tcp4 127.0.0.2:46865: getsockopt: connection timed out

File: pkg/tracer/offsetguess.go [hunk 1]
```diff
@@ -302,18 +316,18 @@ func guess(b *elf.Module) error {
 	}
 
 	listenCompleted := make(chan freePort)
-	closeListen := make(chan struct{})
+	acceptChan := make(chan struct{})
 
 	var listenPort uint16
-	go listenV4(listenCompleted, closeListen)
+	go listenV4(listenCompleted, acceptChan)
```
Comment (2opremio) [inline pkg/tracer/offsetguess.go]: I should have noticed this earlier but, how about splitting the listening and accepting parts? (Making the listening part blocking) Using a channel just to immediately block on it just complicates things

File: pkg/tracer/offsetguess.go [hunk 2]
```diff
@@ -69,7 +69,7 @@ type fieldValues struct {
 	daddrIPv6 [4]uint32
 }
 
-func listenV4(listenCompleted chan freePort, closeListen chan struct{}) {
+func listenV4(listenCompleted chan freePort, acceptChan chan struct{}) {
```
Comment (2opremio) [inline pkg/tracer/offsetguess.go]: `acceptChan` should probably just be called `stop`

**Comment by iaguis** [comment]
Updated.

**Review by 2opremio — APPROVED** [review]

**Pull request merged**